### PR TITLE
Use typed HRW methods

### PR DESCRIPTION
**Pull request by dstepanov-yadro**
Update HRW pkg and use typed HRW methods to sort nodes

### CODE REVIEW

File: netmap/netmap.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ func (m NetMap) PlacementVectors(vectors [][]NodeInfo, pivot []byte) ([][]NodeIn
 	for i := range vectors {
 		result[i] = make([]NodeInfo, len(vectors[i]))
 		copy(result[i], vectors[i])
-		hrw.SortSliceByWeightValue(result[i], nodes(result[i]).weights(wf), h)
+		hrw.SortHasherSliceByWeightValue(result[i], nodes(result[i]).weights(wf), h)
```
Comment (fyrchik) [inline netmap/netmap.go:155]: Can we do an optimization with hash precalculation, similar to what you've done in the `frostfs-node`?

Should be done in `readFromV2` and `SetPublicKey` functions. We should also fallback to a normal calculation if the hash is 0. Am I missing something here? @carpawell

Comment (dstepanov-yadro) [reply netmap/netmap.go:155]: done

Comment (fyrchik) [reply netmap/netmap.go:155]: Could you mention about this in the commit message? Actually, a separate commit for this specific opitimization would be nice.

**Review by acid-ant — DISMISSED** [review]

File: netmap/netmap.go (hunk 1 above)
Comment (dstepanov-yadro) [reply netmap/netmap.go:155]: done

**Review by acid-ant — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]

**Pull request merged**